python-sgd-regression/sgd_regression.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+# QUESTION: when creating files in python-sgd-regression I copied folder python-linear-regression and just renamed
+# `linear-regression` to `sgd-regression`. Is there another way? e.g. cookiecutter template or something?
+
+# QUESTION: I'd like to create some python unit tests in py.test (to speed up development), any suggestions where should
+# I put them?
+
```

Comment:
I'm starting to add unit tests as well. Check this: https://github.com/LREN-CHUV/algorithm-repository/tree/feature/test-histograms/python-histograms/tests

Reply:
Oh nice. I'll reuse it in my code. Btw have you tried [py.test](https://docs.pytest.org/en/latest/) instead of python unittest library? It has made testing & development so much easier and enjoyable for me. There's just so much boilerplate code in standard python unittests.

Reply:
We are using Nose (https://nose.readthedocs.io/). It also claims to make testing easier...

Reply:
Can I try pytest first if you don't mind and in case you don't like it, switch to nose?